payroll/models/hr_payslip_run.py

```diff
@@ -30,6 +30,13 @@ class HrPayslipRun(models.Model):
         tracking=1,
         default="draft",
     )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        required=True,
+        copy=False,
+        default=lambda self: self.env.company,
+    )
```

Comment:
Please also add the accompanying xml view field properly restricted to users with multi-company access. There are plenty of examples in other modules if you need help.

Reply:
Yes, right to fix it.

Thanks for your comments.